Build/PreBuild.ps1

```diff
@@ -70,10 +78,42 @@ $outdir = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPat
 Push-Location "$buildroot\Build"
 try {
 
-    # Install pylance version specified in package.json
+    "Installing Pylance"
+
+    # Install the specified pylance version.
+    # See https://github.com/microsoft/pyrx/wiki/Pylance-release-process#versioning for info about how pylance is versioned and released.
+
     # If this doesn't work, you probably need to set up your .npmrc file or you need permissions to the feed.
     # See https://microsoft.sharepoint.com/teams/python/_layouts/15/Doc.aspx?sourcedoc=%7B30d33826-9f98-4d3e-890e-b7d198bbbcbe%7D&action=edit&wd=target(Python%20VS%2FDev%20Docs.one%7Cd7206ce2-cf40-437b-8ce9-1e55f4bc2f44%2FPylance%20in%20VS%7C6000d391-4e62-4a4d-89d2-7f7c1f005639%2F)&share=IgEmONMwmJ8-TYkOt9GYu7y-AeCM6R8r8Myty0Lj8CeOs4E
-    "Installing Pylance"
+
+    # If the specified version is "latest", find the latest release of pylance based on the specified release type.
+    if ($pylanceVersion -eq "latest") {
+        
+        "Pylance version = $pylanceVersion"
+        "Pylance release type = $pylanceReleaseType"
+
+        # Get all the versions in the feed in descending order
+        $versions = npm view @pylance/pylance versions --json | ConvertFrom-Json
+        [array]::Reverse($versions)
+
+        # Find the highest version with an appropriate patch number.
+        # Stable releases have patch numbers < 100, while preview releases have patch numbers >= 100.
+        foreach ($version in $versions) {
+            $patchVersion = $version.Split(".")[2]
+
+            if ($patchVersion -lt 100 -and $pylanceReleaseType -eq "stable") {
+                $pylanceVersion = $version
```

Comment:
I don't much about Powershell, does `Split` give you a string or number? Do you need to convert it to a number before the comparison?

Reply:
It's a normal string split. You give it a delimiter and it splits a string based on that delimiter. Returns an array of the split string, with the delimiter removed.

In this case, we're splitting a version number that looks like this M.m.p and taking the 3rd element from the split, which is the patch number.

Reply:
I wasn't sure if I had to convert the element to an integer before the comparison, so I tried it out. Turns out you don't have to convert it, it does it automatically.

Reply:
I will add in the conversion though, just so it's clear what the intent it 😄

Reply:
Sounds great, thanks for doing this:)